tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -184,6 +185,8 @@ sub find_sriov_ethernet_devices {
         if ((script_run "lspci -v -s $_ | grep -q 'SR-IOV'") == 0) {
             #only those vfs whose pv can be brought up can be passed through to guest vms
             my $nic = script_output "ls -l /sys/class/net |grep $_ | awk '{print \$9}'";
+            #STARTMODE='auto' because it is the only NIC in the SUT in OSD
+            #but it will cause trouble for running test on local machines with multiple NICs without fresh installation
             script_run "echo \"BOOTPROTO='dhcp'\nSTARTMODE='auto'\" > /etc/sysconfig/network/ifcfg-$nic";
```

Comment:
I think that this action does not belong to this routine neither to this file.
Primary NIC should be configured during the installation or directly after.
Other NICs should not have such settings at all.
This has also nothing to do with passthrough.

Reply:
To passthrough a VF to guest from a NIC with SRIOV ability in host, the PF is required to be in UP state. If primary NIC is used, no additional configuration is needed; But if it's another NIC, the procedure is necessary. Only the NICs which is able to be brought up can be used for sriov tests.

Reply:
To passthrough a VF to guest from a NIC with SRIOV ability in host, the PF is required to be in UP state. If primary NIC is used, no additional configuration is needed; But if it's another NIC, the procedure is necessary. Only the NICs which is able to be brought up can be used for sriov tests.

Reply:
That is true, but we should differentiate between main and the other NICs. The primary NIC should not be touched and keep its settings and the others should not get an IP from DHCP as that will happen in the guest.

Reply:
I am not quite familiar with NICs, Is it appropriate to search the 'master' keyword in nic state to distinguish primary with other NICs?

about assigning IP, it is different between normal NICs and the NICs with SRIOV capability. In the case of passing through a normal NIC, it need not get a IP first on the host. But for a SRIOV NIC, it consist of 1 PF and multiple VFs, the PF must be 'up' then the VFs can be attached to guests. So the SRIOV NIC has to get an IP for the PF, VFs will get ips in the guest only.

Reply:
I am not quite familiar with NICs, Is it appropriate to search the 'master' keyword in nic state to distinguish primary with other NICs?

about assigning IP, it is different between normal NICs and the NICs with SRIOV capability. In the case of passing through a normal NIC, it need not get a IP first on the host. But for a SRIOV NIC, it consist of 1 PF and multiple VFs, the PF must be 'up' then the VFs can be attached to guests. So the SRIOV NIC has to get an IP for the PF, VFs will get ips in the guest only.

Reply:
Primary interface is the interface openQA uses to SSH to the host, so this one should not be touched.

If you pass normal NIC from the guest to the host then it does not have to have IP on the host.

PF can be UP and have no IP address.

Reply:
how can PF be 'up' without an IP address? by using other protocol instead of 'dhcp'? I never thought it could be done that way.

Reply:
```
apu:~ # ip link show enp3s0
4: enp3s0: <BROADCAST,MULTICAST> mtu 1500 qdisc mq state DOWN mode DEFAULT group default qlen 1000
    link/ether 00:0d:b9:57:04:6e brd ff:ff:ff:ff:ff:ff
apu:~ # ip address show enp3s0
4: enp3s0: <BROADCAST,MULTICAST> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether 00:0d:b9:57:04:6e brd ff:ff:ff:ff:ff:ff
apu:~ # ip link set up enp3s0
apu:~ # ip link show enp3s0
4: enp3s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN mode DEFAULT group default qlen 1000
    link/ether 00:0d:b9:57:04:6e brd ff:ff:ff:ff:ff:ff
apu:~ # ip address show enp3s0
4: enp3s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether 00:0d:b9:57:04:6e brd ff:ff:ff:ff:ff:ff
```
You may put something like this to you `/etc/sysconfig/network/ifcfg-eth1`:
```
BOOTPROTO=none
ONBOOT=yes
```
With that we are sure that both host and guests have only addresses we are aware of.

Reply:
"BOOTPROTO=none" works! thanks, Pavel. And I find a way to determine the primary NIC in OSD. Verifying...